### PR TITLE
Ex UI 1112 new branch update hearing requirements event is stuck

### PR DESCRIPTION
**Pull request by OgunyemiO**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EXUI-1112

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by olusegz07 — DISMISSED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 1]
```diff
@@ -309,10 +309,16 @@ export class FieldsPurger {
    * @returns A new array with the mapped values
    */
   public mapArrayValuesToNull(array: any[]): any[] {
-    return array.map(element => {
-      return typeof element === 'object'
-        ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
-        : null;
-    });
+    if (array && array.length > 0) { FieldsUtils.isArray([])
```
Comment (RiteshHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:312]: @OgunyemiO why you have added FieldsUtils.isArray([]?

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 2]
```diff
@@ -309,10 +309,16 @@ export class FieldsPurger {
...
+      return array.map(element => {
+        if (element !==  undefined  && element !== null) {
+          return typeof element === 'object'
+            ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
+            : null;
+        }
```
Comment (RiteshHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:318]: @OgunyemiO are you not checking element if it is null or undefined?

**Review by richardhenryash — APPROVED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts (hunk 1 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:312]: @RiteshHMCTS Oops! I was trying something out and mistakenly left it. :) I will remove now.

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts (hunk 2 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:318]: @RiteshHMCTS  The check is there:

 if (element !==  undefined  && element !== null) {
          return typeof element === 'object'
            ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
            : null;
        }

Comment (RiteshHMCTS) [reply projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:318]: @OgunyemiO there is if condition for element is not undefined or not null but no else condition

**Review by RiteshHMCTS — COMMENTED** [review]
@OgunyemiO also you have not updated toolkit version

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by richardhenryash — APPROVED** [review]

**Review by andywilkinshmcts — DISMISSED** [review]

**Pull request merged**